webalyzer/collector/static/collect.js

```diff
@@ -42,44 +53,78 @@
     };
     req.send();
   }
+
   function postCSS(url, callback) {
     _getCSS(url, function(status, response) {
       if (status === 200) {
-        var req = request();
-        var data = new FormData();
-        data.append('url', url);
-        data.append('domain', domain);
-        data.append('css', response);
-        _post(data, function(status) {
-          if (status === 200) {
-            console.log("We have already collected this CSS");
-            if (callback) callback(url, false);
-          } else if (status === 201) {
-            console.log("Yum! Collected another chunk of CSS");
-            if (callback) callback(url, true);
-          } else {
-            console.warn("Failed to collect the CSS of this page");
-          }
+        var source_hash = makeHash(response);
+
+        _postHash(source_hash, 'css', function(status) {
+            if(status === 200) {
+                console.log('We have already collected CSS matching ' + source_hash);
+            } else {
+                console.log('Sending CSS.');
+                var req = request();
+                var data = new FormData();
+                data.append('url', url);
+                data.append('domain', domain);
+                data.append('css', response);
+                data.append('source_hash', source_hash);
+                _post(data, function(status) {
+                    if (status === 200) {
+                        console.log("We have already collected this CSS");
+                    if (callback) callback(url, false);
+                    } else if (status === 201) {
+                        console.log("Yum! Collected another chunk of CSS");
+                    if (callback) callback(url, true);
+                    } else {
+                        console.warn("Failed to collect the CSS of this page");
+                    }
+                });
+            };
         });
       }
     });
   }
 
+  // hashCode() from http://erlycoder.com/49/javascript-hash-functions-to-convert-string-into-integer-hash-
+  function makeHash(source) {
+    var hash = 0;
+    if (source.length == 0) return hash;
+    for (i = 0; i < source.length; i++) {
+        char = source.charCodeAt(i);
+        hash = ((hash<<5)-hash)+char;
+        hash = hash & hash; // Convert to 32bit integer
+    }
+    return hash;
+  }
```

Comment:
Cool! This looks really fast. 
It's a very very short hash (few combinations) but I guess when you combine it with the URL and/or the domain it doesn't matter.